app/views/shared/_alert.html.erb

```diff
@@ -0,0 +1,11 @@
+<% type = local_assigns.fetch(:type, 'other') %>
+<% classes = 'usa-alert usa-alert--' + type %>
+<% classes += ' ' + local_assigns[:class] if local_assigns[:class] %>
+<% role = type === 'error' ? 'alert' : 'status' %>
+<% message = yield.presence || local_assigns.fetch(:message) %>
+
+<%= tag.div class: classes, role: role do %>
```

Comment:
Two suggestions:

1. use one big `<% .. %>` instead of having to do each line
2. let's build the classes as an array? I hate having to worry about spacing
3. I think calling `yield` without a block should error...? maybe this is some ERB magic... anyways I'd check for `block_given?` if this were "normal" ruby. Unsure if this is needed but I'll write it anyways


```suggestion
<%
  type = local_assigns.fetch(:type, 'other')
  role = type === 'error' ? 'alert' : 'status' 

  classes = [
    'usa-alert',
    "usa-alert--#{type}",
  ]
  classes << local_assigns[:class] if local_assigns[:class]
  message = block_given? ? yield : local_assigns.fetch(:message) 
%>

<%= tag.div class: classes, role: role do %>
```

Reply:
> Two suggestions:
> 
> 1. [...]
> 2. [...]
> 3. [...]

This reminds me of: https://twitter.com/secretGeek/status/7269997868 😄 

But all good solutions!

Re: `yield`, I'd found a few unofficial resources which seemed to indicate this approach was valid:

- https://stackoverflow.com/a/22651804
- https://goiabada.blog/rails-components-faedd412ce19

The latter article also has a relevant blurb w.r.t. your `block_given?` suggestion:

>In this particular case of rendering a partial with a block without the `layout` option, we cannot simply check that a block has been used with `block_given?` inside the partial. It returns `true` and `yield` returns `""`. That's why we check if the block has a real content with `presence`.

I'd like (yet struggle to find) a more authoritative resource, but in any case, I think our test cases give us sufficient coverage if absence of a block was a problem? (specifically `it 'renders message from param'`)

Reply:
I also kinda wondered if supporting _both_ block and string variants might be getting a bit too much into [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) territory, if it'd make it easier to address this by just removing support for block content 🤷

Reply:
> * let's build the classes as an array? I hate having to worry about spacing

TIL the `tag` helper supports `class` given as an array. Handy :+1:

Reply:
Updates in 8f2427f and 65adc37.

Reply:
TIL tag helper supports an array too! I had just forgotten the `classes.join(' ')` lol